File: C++/Interpreter/BooleanExpressions/context.cc
#include <iostream>
#include "context.h"
#include "boolean_expression.h"

Context::Context() : _variable_assignments() {}

void Context::assign(VariableExpression* variable, bool value) {
    _variable_assignments[variable->get_name()] = value;

}

bool Context::lookup(const std::string& name) const {
    return _variable_assignments.at(name);
}

void Context::show_assignments() const {
    std::cout << "Boolean values for the variables are:\n";
    for (auto& [key, val] : _variable_assignments) {
        std::cout << key << " = " << val << "\n";
    }
}

File: Java/Interpreter/BooleanExpressions/src/com/ericgtkb/Context.java
package com.ericgtkb;

import java.util.HashMap;
import java.util.Map;

public class Context {
    private Map<String, Boolean> variableAssignments;

    public Context() {
        variableAssignments = new HashMap<>();
    }

    public void assign(VariableExpression variable, boolean value) {
        variableAssignments.put(variable.getName(), value);
    }

    public boolean lookup(String name) {
        return variableAssignments.get(name);
    }

    public void showAssignments() {
        System.out.println("Boolean values for the variables are:");
        for (var entry : variableAssignments.entrySet()) {
            System.out.println(entry.getKey() + " = " + entry.getValue());
        }
    }
}
